src/main/java/build/buildfarm/worker/Executor.java

```diff
@@ -199,11 +210,13 @@ public void run() {
         if (remainingNanoTime > 0) {
           if (process.waitFor(remainingNanoTime, TimeUnit.NANOSECONDS)) {
```

Comment:
I don't really understand this line, btw -- if process.waitFor returns false, doesn't that mean the time has elapsed fully and we should be DEADLINE_EXCEEDED already? And then why do we need a loop?

Reply:
My C is showing (expecting EINTR here as a possible response to `wait(2)`). You're right, documentation indicates that if we don't get the waitFor true return, we will get false or InterruptedException will be thrown.